### PR TITLE
implement a tool for verifying language grammars

### PR DESCRIPTION
**Pull request by zerbina**
The `passtool` currently only performs some basic checks (i.e., whether
all references name existing rules and that no duplicate productions
are introduced).

In the future, it'll also provide code-generation facilities for
automating the creation of validation layers for languages. The idea is
to reduced the overhead of removing, changing, or adding new
intermediate languages as much as possible.

The tool is ran as part of CI to make sure all language grammars are
valid.

### CODE REVIEW

**Review by saem — APPROVED** [review]
This looks great!

Wow, I thought this tool might have been out of reach for a while, but this is really nice. 🚀

**Comment by saem** [comment]
I didn't merge it in case there was more to it than waiting for the merge of the other PR, apologies if that was the only blocker.

File: tools/passtool/parser.nim [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+    tkParensLe, tkParensRi
+    tkDot
+    tkIdent
+    tkColon
+    tkEq
+    tkPlus
+    tkMinusEq
+    tkQMark
+    tkStar
+    tkSharpLe, tkSharpRi
```
[the remark this answers is not in the document]
Comment (zerbina) [reply tools/passtool/parser.nim:21]: Ah, thanks. I temporarily used "sharp" because "angle" eluded me, but then I forgot to update the enum.

**Comment by zerbina** [comment]
> I didn't merge it in case there was more to it than waiting for the merge of the other PR, apologies if that was the only blocker.

No no, it's all good. The reason I wanted to wait with merging was so that I can update CI to test `lang1` instead of `lang0`.

**Pull request merged**